diff --git sites/all/modules/contrib/s3fs/s3fs.module sites/all/modules/contrib/s3fs/s3fs.module
index d4dcc3a..90df9b9 100644
--- sites/all/modules/contrib/s3fs/s3fs.module
+++ sites/all/modules/contrib/s3fs/s3fs.module
@@ -649,7 +649,7 @@ function _s3fs_convert_metadata($uri, $s3_metadata) {
       $metadata['timestamp'] = date('U', strtotime($s3_metadata['LastModified']));
     }
 
-    if (isset($s3_metadata['VersionId'])) {
+    if (isset($s3_metadata['VersionId']) && $s3_metadata['VersionId'] != 'null') {
       $metadata['version'] = $s3_metadata['VersionId'];
     }
   }
