diff --git a/includes/common.inc b/includes/common.inc
index c97704c..a0ee9e2 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -4396,7 +4396,7 @@ function drupal_get_js($scope = 'header', $javascript = NULL, $skip_alter = FALS
   // Filter out elements of the given scope.
   $items = array();
   foreach ($javascript as $key => $item) {
-    if ($item['scope'] == $scope) {
+    if (array_key_exists('scope', $item) && $item['scope'] == $scope) {
       $items[$key] = $item;
     }
   }
